configs/src/types.ts

```diff
@@ -12,5 +14,39 @@ export type NetworkAliases = {
 };
 
 export type NetworkConfigs<T = NetworkConfig> = {
-  [network: string]: T;
+  [network in SupportedNetworks]: T;
+};
+
+export enum SupportedNetworks {
+  MAINNET = 'mainnet',
+  GOERLI = 'goerli',
+  SEPOLIA = 'sepolia',
+  POLYGON = 'polygon',
+  MUMBAI = 'mumbai',
+  BASE = 'baseMainnet',
+  BASE_GOERLI = 'baseGoerli',
+  BASE_SEPOLIA = 'baseSepolia',
+  ARBITRUM = 'arbitrum',
+  ARBITRUM_SEPOLIA = 'arbitrumSepolia',
+}
+
+// the entries in this enum has to be in order from
+// oldest to newest so that getLatestNetworkVersion() works as expected
```

Comment:
Can we write a test to make this foolproof?

Reply:
puhh I would need to write a parser to parse the version number to make sure that it is in the correct order. you think people cannot keep this rule?

Reply:
Yes, should be fine